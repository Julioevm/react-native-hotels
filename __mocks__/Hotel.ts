export const Hotel = {
  id: '1',
  name: 'Hotel',
  stars: 4,
  userRating: 4.5,
  price: 100,
  currency: 'EUR',
  location: {
    latitude: 0,
    longitude: 0,
    city: 'Madrid',
    country: 'Spain',
    address: 'Calle de la Princesa, 1',
  },
  checkIn: {from: '2020-01-01', to: '2020-01-02'},
  checkOut: {from: '2020-01-01', to: '2020-01-02'},
  contact: {
    phoneNumber: '123456789',
    email: 'test@mail.com',
  },
  gallery: ['https://picsum.photos/200/300'],
};
