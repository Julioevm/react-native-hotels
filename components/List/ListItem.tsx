import {Image, StyleSheet, Text, TouchableOpacity, View} from 'react-native';
import React from 'react';
import StarRating from '../StarRating';
import UserRating from '../UserRating';

interface Props {
  name: string;
  stars: number;
  score: number;
  price: number;
  gallery: Array<string>;
  currency: string;
}

export default function ListItem(props: Props) {
  const {name, stars, score, price, currency} = props;

  return (
    <TouchableOpacity>
      <View style={styles.container}>
        <View>
          <Image style={styles.image} source={{uri: props.gallery[0]}} />
          <Text style={styles.name}>{name}</Text>
          <View style={styles.ratings}>
            <StarRating stars={stars} />
            <UserRating rating={score} />
          </View>
          <Text style={styles.price}>{`${price} ${currency}`}</Text>
        </View>
      </View>
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  container: {
    paddingVertical: 8,
    paddingHorizontal: 16,
  },
  image: {
    width: '100%',
    height: 120,
    resizeMode: 'cover',
  },
  ratings: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  name: {
    fontSize: 16,
    fontWeight: 'bold',
  },
  price: {
    textAlign: 'right',
    fontSize: 16,
    fontWeight: 'bold',
  },
});
