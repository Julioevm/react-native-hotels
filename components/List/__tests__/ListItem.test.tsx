import {useNavigation} from '@react-navigation/native';
import {render, fireEvent} from '@testing-library/react-native';
import {Hotel} from '../../../__mocks__/Hotel';
import ListItem from '../ListItem';

jest.mock('@react-navigation/native', () => {
  return {
    useNavigation: jest.fn(),
  };
});

describe('ListItem', () => {
  it('navigates to the "Details" route when pressed', () => {
    const navigation = {
      navigate: jest.fn(),
    };
    (useNavigation as jest.Mock).mockReturnValue(navigation);

    const {getByText} = render(<ListItem hotel={Hotel} />);

    fireEvent.press(getByText('Hotel'));

    expect(navigation.navigate).toHaveBeenCalledWith('Details', {
      hotel: Hotel,
    });
  });
});
