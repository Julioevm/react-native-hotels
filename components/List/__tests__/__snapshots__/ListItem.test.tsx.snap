// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem should render a ListItem 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "white",
      "borderRadius": 8,
      "margin": 16,
      "overflow": "hidden",
      "paddingBottom": 8,
    }
  }
  testID="list-item"
>
  <View>
    <Carrousel
      gallery={
        [
          "https://res.cloudinary.com/lastminute/image/upload/t_OSE_redes_item_view/v1499779963/Swindon_yjsrwz.jpg",
        ]
      }
      width={718}
    />
    <View
      style={
        {
          "paddingHorizontal": 16,
          "paddingVertical": 8,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
        testID="list-item-name"
      >
        Park Plaza London Waterloo
      </Text>
      <View
        style={
          {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginVertical": 8,
          }
        }
      >
        <View
          style={
            {
              "flexDirection": "row",
            }
          }
        >
          <Image
            source={
              {
                "testUri": "../../../assets/star-small.png",
              }
            }
            style={
              [
                {
                  "height": 16,
                  "width": 16,
                },
                {
                  "tintColor": "orange",
                },
              ]
            }
          />
          <Image
            source={
              {
                "testUri": "../../../assets/star-small.png",
              }
            }
            style={
              [
                {
                  "height": 16,
                  "width": 16,
                },
                {
                  "tintColor": "orange",
                },
              ]
            }
          />
          <Image
            source={
              {
                "testUri": "../../../assets/star-small.png",
              }
            }
            style={
              [
                {
                  "height": 16,
                  "width": 16,
                },
                {
                  "tintColor": "orange",
                },
              ]
            }
          />
          <Image
            source={
              {
                "testUri": "../../../assets/star-small.png",
              }
            }
            style={
              [
                {
                  "height": 16,
                  "width": 16,
                },
                {
                  "tintColor": "orange",
                },
              ]
            }
          />
          <Image
            source={
              {
                "testUri": "../../../assets/star-small.png",
              }
            }
            style={
              [
                {
                  "height": 16,
                  "width": 16,
                },
                {
                  "tintColor": "gray",
                },
              ]
            }
          />
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Text>
            User Rating
          </Text>
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "borderRadius": 4,
                  "height": 18,
                  "justifyContent": "center",
                  "marginLeft": 6,
                  "width": 18,
                },
                {
                  "backgroundColor": "hsl(108, 100%, 70%)",
                },
              ]
            }
          >
            <Text
              adjustsFontSizeToFit={true}
              style={
                {
                  "fontWeight": "bold",
                  "textAlign": "center",
                  "textAlignVertical": "center",
                }
              }
            >
              9
            </Text>
          </View>
        </View>
      </View>
      <Text
        style={
          {
            "fontSize": 18,
            "fontWeight": "bold",
            "textAlign": "right",
          }
        }
      >
        120 â‚¬
      </Text>
    </View>
  </View>
</View>
`;
